cumulus/client/cirrus-executor/src/processor.rs

```diff
@@ -195,28 +195,27 @@ where
 			self.runtime_api.runtime_api().intermediate_roots(&BlockId::Hash(parent_hash))?;
 		roots.push(state_root.encode());
 
+		let trace_root =
+			BlakeTwo256::ordered_trie_root(roots.clone(), sp_core::storage::StateVersion::V1);
```

Comment:
Hmm, what's a regular Merkle Tree? And the suggestion here is?

Reply:
Merkle Tree is Merkle Tree. Merkle Particia Trie (that you used there) is a different (but related) data structure.

Reply:
Sure, I see the point. Usually, I mix them up in practice unless it's necessary to distinguish because you could use both of them for some common purposes. So, you want it to be a plain Merkle Tree not a MPT here?

Reply:
Yes, that was the design we went for unless I missed something. Do we have other places where we use MPT for similar purpose?

Reply:
This is the only one for now. Does using MPT have some downsides? Perhaps we could use it since substrate already provides this convenient API if there is no harm doing so.

Reply:
MPT is a different data structure used for storing data in it. MT is not. We don't need some ordered version of that with layout V1, we need a simple MT instead. I understand that we have this convenient API to get some root, but it is not what we need.

Reply:
Please take a look at https://github.com/subspace/subspace/pull/236/commits/e0631d3df11b847135019ae4bf64a70e548903fe